Remotion/Validation/UnitTests/RuleBuilders/AddingObjectValidationRuleBuilderTest.cs

```diff
@@ -14,61 +14,153 @@
+      Assert.That(_addingObjectValidationBuilder.ObjectMetaValidationRuleCollector, Is.SameAs(_objectMetaValidationRuleCollectorMock.Object));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void SetValidator ()
     {
+      Func<ObjectValidationRuleInitializationParameters, IObjectValidator> validatorFactory = _ => _objectValidatorStub.Object;
+      _addingObjectValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterValidator(validatorFactory))
+          .Verifiable();
+
+      ((IAddingObjectValidationRuleBuilder<Customer>)_addingObjectValidationBuilder).SetValidator(validatorFactory);
+
+      _addingObjectValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CanBeRemoved ()
     {
+      _addingObjectValidationRuleCollectorMock
+          .Setup(mock => mock.SetRemovable())
+          .Verifiable();
+
+      _addingObjectValidationBuilder.CanBeRemoved();
+
+      _addingObjectValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule ()
     {
+      var metaValidationRuleStub = new Mock<IObjectMetaValidationRule>();
+
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(metaValidationRuleStub.Object))
+          .Verifiable();
+
+      _addingObjectValidationBuilder.AddMetaValidationRule(metaValidationRuleStub.Object);
+
+      _objectMetaValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_FuncOverload ()
     {
+      var fakeValidationResult = MetaValidationRuleValidationResult.CreateValidResult();
+
+      IObjectMetaValidationRule rule = null;
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(It.IsAny<IObjectMetaValidationRule>()))
+          .Callback((IObjectMetaValidationRule r) => rule = r);
+
+      _addingObjectValidationBuilder.AddMetaValidationRule(v => fakeValidationResult);
+
+      Assert.That(rule.GetType(), Is.EqualTo(typeof(DelegateObjectMetaValidationRule<IObjectValidator>)));
+      Assert.That(rule.Validate(new IObjectValidator[0]).First(), Is.EqualTo(fakeValidationResult));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_ExpressionOverload_IsValidIsTrue ()
     {
+      IObjectMetaValidationRule rule = null;
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(It.IsAny<IObjectMetaValidationRule>()))
+          .Callback((IObjectMetaValidationRule r) => rule = r);
+
+      _addingObjectValidationBuilder.AddMetaValidationRule<IObjectValidator>(v => true);
+
+      Assert.That(rule.GetType(), Is.EqualTo(typeof(DelegateObjectMetaValidationRule<IObjectValidator>)));
+      Assert.That(rule.Validate(new IObjectValidator[0]).First().IsValid, Is.True);
```

Comment:
This one's a bit trickier. We fully expect that there's just one item, so we could at least use Single instead of First. Alternatively, and possibly nicer, is that we can just go an do the following:
```suggestion
      Assert.That(rule.Validate(new IObjectValidator[0]), Has.Exactly(1).With.Property(nameof(MetaValidationRuleValidationResult.IsValid)).True);
```